src/loaders/createLoaders.ts

```diff
@@ -98,7 +124,9 @@ export function createLoaders(user?: UserDocument): Loaders {
           modelIds,
           ModelModel.find({
             _id: { $in: Array.from(modelIds) },
-            ownerId: user?._id,
+            // ownerId: user?._id,
+            // In order to get data for the Deck at the marketplace
+            // it is necessary get the model without being the model owner
```

Comment:
can we create another loader for this? because this way we would "enable" users to use the current model query to get models from a different user

Reply:
I tried going this route, but I am not calling this method directly. 

It is called on:

- `src/resolvers/deck/types.ts` in the `NoteType` definition
- `src/resolvers/field/types.ts` in the `FieldType` definition
- `src/resolvers/model/addTemplateToModel.ts` in the `addTemplateToModel` resolver
- many other places

The problem with this approach is that it will mean that I have to create many new definitions. For example with the `NoteType` definition. It calls `deckLoader` and `modelLoader` so I'll need to create another definition for `PublishedDeckType` `PublishedNoteType` `PublishedFieldType` `PublishedFieldTemplate` and do a lot of duplication in order to use the correct loaders.

Maybe another solution is to add a `published` field into models, templates, notes, fields that are used in a published deck. Then the loader query can be:

```typescript
          ModelModel.find({
            _id: { $in: Array.from(modelIds) },
           $or:[{ownerId: user?._id},{published:true}]
          }),
```

If you want I can make the necessary changes so it works creating new loader/type definitions. Just let me know

Reply:
hmmm I see. Can you add a `TODO` in the comment so I remember to fix it later?